cardano-cli/src/Cardano/CLI/Compatible/Transaction.hs

```diff
@@ -294,8 +295,8 @@ runCompatibleTransactionCmd
 
         proposalsRefInputs =
           [ refInput
-          | ProposalProcedures _ (TxProposalProcedures _ (BuildTxWith proposalMap)) <- [protocolUpdates]
-          , sWit <- Map.elems proposalMap
+          | ProposalProcedures _ (TxProposalProcedures proposalMap) <- [protocolUpdates]
+          , BuildTxWith (Just sWit) <- map snd $ OMap.toAscList proposalMap
```

Comment:
Can you explain the reason behind this change?

Reply:
This comes from here: https://github.com/IntersectMBO/cardano-api/pull/726

Reply:
`toAscList` is incorrect here, changes the order of witnesses. Tbh I don't remember if that matters for the reference inputs.

Reply:
It should be
```haskell
, BuildTxWith (Just sWit) <- map snd $ toList proposalMap
```

Reply:
Ah, it would be `asocs` then. I thought OMap only had `Ord` orderer. Nice catch

Reply:
Addressed here: https://github.com/IntersectMBO/cardano-cli/pull/1068/commits/ed8086df3ee91f8834d88d635cbfb121fa173021